meta-language.md

```diff
@@ -865,34 +933,138 @@ The equality type is computed with
 
     c₁ ≡ c₂
 
-Instead of the character `≡` you may use `==`.
+Instead of the character `≡` you may use `==`. There are a number of constructors for
+generating elements of the equality types, as follows.
 
-#### Reflexivity
+###### Reflexivity
 
 The reflexivity term is computed with
 
     refl c
 
-#### Equality checking
+If `c` evaluates to $\isterm{\G}{\e}{\tyA}$ then `refl c` evaluates to $\isterm{\G}{\juRefl{\tyA} \e}{\JuEqual{\tyA}{\e}{\e}}$.
 
-TODO: describe how equality checking works by invocation of operations
+##### Reduction
 
-##### Congruences
+The rule [`prod-beta`](./type-theory.html#prod-beta) is available through
 
-TODO: Describe `congruence`
+    beta_step x A B e₁ e₂
 
-##### Extensionality
+where:
 
-TODO: Describe `extensionality`
+* `A` evaluates to a type $\istype{\G}{\tyA}$
```

Comment:
No.
x evaluates to x : A1
A evaluates to evidence that A1 = A2
B evaluates to evidence that x : A1 |- B1 = B2
e1 evaluates to x : A1 |- e1 : B1
e2 evaluates to |- e2 : B2
(no checking mode is used to evaluate e1 and e2 unlike what might be expected)

then we get (lambda x : A1, e1) @[x : A2, B2] e2 = e1[e2/x]